subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputIntegrationTest.groovy

```diff
@@ -153,8 +153,7 @@ class StaleOutputIntegrationTest extends AbstractIntegrationSpec {
                 doLast {
                     assert !file("build/file").exists()
                     file("build/file").text = "Created"
-                    assert !file("build/dir").exists() 
```

Comment:
❌ Same as above. We would need to create a file in `build/dir` and then check that this file has been removed. If not, we do not test any more that directories are being removed. It would be nice to have a way to check that the root directory has been removed, too, but I guess we can't :(

Reply:
Added some stale files.

Reply:
And a check that the directory is empty.